src/main/java/ch/njol/skript/sections/recipe/SecCookingRecipe.java

```diff
@@ -0,0 +1,179 @@
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult, SectionNode sectionNode, List<TriggerItem> triggerItems) {
+		EntryContainer entryContainer = validator.validate(sectionNode);
+		if (entryContainer == null)
+			return false;
+		type = CookingRecipeType.values()[parseResult.mark];
+		ingredient = (Expression<? extends ItemType>) entryContainer.get("ingredient", false);
+		cookTime = (Expression<? extends Timespan>) entryContainer.get("cook time", false);
+		xp = (Expression<? extends Number>) entryContainer.get("xp", false);
+		group = (Expression<? extends String>) entryContainer.getOptional("group", false);
+		result = (Expression<? extends ItemType>) entryContainer.get("result", false);
+		key = (Expression<? extends String>) entryContainer.get("key", false);
+		return true;
+	}
+
+	@Override
+	public String toString(Event event, boolean debug) {
+		String friendlyName = type.name().replace("_", " ").toLowerCase();
+		return "create a " + friendlyName + " recipe";
+	}
+
+	@Override
+	protected TriggerItem walk(Event event) {
+		execute(event);
+		return super.walk(event, false);
+	}
+
+	private void execute(Event event) {
+		String key = this.key.getSingle(event);
+		if (key == null || StringUtils.isBlank(key))
+			return;
+		ItemType result = this.result.getSingle(event);
+		if (result == null)
+			return;
+		ItemType ingredient = this.ingredient.getSingle(event);
+		if (ingredient == null)
+			return;
+		Number xp = this.xp.getSingle(event);
+		if (xp == null)
+			return;
+		Timespan cookTime = this.cookTime.getSingle(event);
+		if (cookTime == null)
+			return;
+
+		Set<Material> ingredientMaterials = new HashSet<>();
+		for (ItemStack itemStack : ingredient.getAll()) {
+			ingredientMaterials.add(itemStack.getType());
+		}
+
+		// the recipe APIs require an int :(
+		int cookTimeTicks = (int) cookTime.getTicks_i();
+		NamespacedKey namespacedKey = Utils.createNamespacedKey(key);
+		RecipeChoice choice = new RecipeChoice.MaterialChoice(ingredientMaterials.toArray(new Material[0]));
+		CookingRecipe<?> recipe;
+		switch (type) {
+			case SMOKER:
+				recipe = new SmokingRecipe(namespacedKey, result.getRandom(), choice, xp.floatValue(), cookTimeTicks);
+				break;
+			case FURNACE:
+				recipe = new FurnaceRecipe(namespacedKey, result.getRandom(), choice, xp.floatValue(), cookTimeTicks);
+				break;
+			case CAMPFIRE:
+				recipe = new CampfireRecipe(namespacedKey, result.getRandom(), choice, xp.floatValue(), cookTimeTicks);
+				break;
+			case BLAST_FURNACE:
+				recipe = new BlastingRecipe(namespacedKey, result.getRandom(), choice, xp.floatValue(), cookTimeTicks);
+				break;
+			default:
+				throw new IllegalStateException();
+		}
+
+		if (this.group != null) {
+			String group = this.group.getSingle(event);
+			if (group != null)
+				recipe.setGroup(group);
+		}
+
+		try {
+			Bukkit.getServer().addRecipe(recipe);
```

Comment:
check this? it seems to return a boolean not throw an exception now that i look again